src/bin/add/args.rs

```diff
@@ -221,7 +230,11 @@ impl Args {
             } else {
                 if let Some(path) = &self.path {
                     assert!(self.registry.is_none());
-                    dependency = dependency.set_path(dunce::canonicalize(path)?);
+                    let dep_path = dunce::canonicalize(path)?;
+                    let manifest = get_manifest_from_path(&dep_path)?;
```

Comment:
ditto about dunce

Reply:
I'm actually not sure how you could move `dunce` into the `parse_crate_name_from_uri` since the uses of dunce aren't actually in the URI branch? they're in the `else` branch, but maybe I'm misunderstanding. Can you elaborate?

Reply:
I was just referring to limiting where the canonicalized path is used
```rust
let manifest = get_manifest_from_path(path)?;
dependency = dependency.set_available_features(manifest.features());
dependency = dependency.set_path(dunce::canonicalize(path)?);
```

Reply:
right, gotcha